core/src/main/com/dacubeking/autobuilder/gui/scripting/RobotCodeData.java

```diff
@@ -31,23 +31,184 @@ public class RobotCodeData {
+        });
+
+        inferableTypesVerification.put(long.class.getName(), s -> {
+            if (!s.matches("^-?[0-9]+$")) {
+                return false;
+            }
+            try {
+                Long.parseLong(s);
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+            return true;
+        });
+
+        inferableTypesVerification.put(short.class.getName(), s -> {
+            if (!s.matches("^-?[0-9]+$")) {
+                return false;
+            }
+            try {
+                Short.parseShort(s);
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+            return true;
+        });
+
+        inferableTypesVerification.put(byte.class.getName(), s -> {
+            if (!s.matches("^-?[0-9]+$")) {
+                return false;
+            }
+            try {
+                Byte.parseByte(s);
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+            return true;
+        });
+
+        inferableTypesVerification.put(byte.class.getName(), s -> {
+            if (!s.matches("^-?[0-9]+$")) {
+                return false;
+            }
+            try {
+                Byte.parseByte(s);
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+            return true;
+        });
+
+        inferableTypesVerification.put(char.class.getName(), s -> s.matches("^.$"));
+
+        inferableTypesVerification.put(boolean.class.getName(), s -> s.matches("^(?i)(true|false)$"));
+
         inferableTypesVerification.put(String.class.getName(), s -> true);
-        inferableTypesVerification.put(Integer.class.getName(), s -> s.matches("[0-9]+"));
-        inferableTypesVerification.put(Double.class.getName(), s -> s.matches("[0-9]+\\.?[0-9]*"));
-        inferableTypesVerification.put(Float.class.getName(), s -> s.matches("[0-9]+\\.?[0-9]*"));
-        inferableTypesVerification.put(Long.class.getName(), s -> s.matches("[0-9]+"));
-        inferableTypesVerification.put(Short.class.getName(), s -> s.matches("[0-9]+"));
-        inferableTypesVerification.put(Byte.class.getName(), s -> s.matches("[0-9]+"));
-        inferableTypesVerification.put(Character.class.getName(), s -> s.matches("[a-zA-Z]+"));
-        inferableTypesVerification.put(Boolean.class.getName(), s -> s.matches("true|false"));
+
+        inferableTypesVerification.put(Integer.class.getName(), s -> {
+            if (!s.matches("^-?[0-9]+$")) {
+                return false;
+            }
+            try {
+                Integer.parseInt(s);
+            } catch (NumberFormatException e) {
+                return false;
+            }
+
+            return true;
+        });
+
+        inferableTypesVerification.put(Double.class.getName(), s -> {
+            if (!s.matches("^[+-]?(\\d+([.]\\d*)?([eE][+-]?\\d+)?|[.]\\d+([eE][+-]?\\d+)?)$")) {
```

Comment:
There seems to be a ton of duplicated code here. Might be useful if you make functions to encapsulate the different repeated validation you do to help with readability/maintainability.

Reply:
just made some changes that reduces a bunch of the duplication